spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/outbound/Mqttv5PahoMessageHandler.java

```diff
@@ -237,10 +265,17 @@ protected void publish(String topic, Object mqttMessage, Message<?> message) {
 		Assert.isInstanceOf(MqttMessage.class, mqttMessage, "The 'mqttMessage' must be an instance of 'MqttMessage'");
 		long completionTimeout = getCompletionTimeout();
 		try {
-			if (!this.mqttClient.isConnected()) {
-				this.mqttClient.connect(this.connectionOptions).waitForCompletion(completionTimeout);
+			IMqttAsyncClient theClient;
+			if (getClientManager() != null) {
+				theClient = getClientManager().getClient();
+			}
+			else {
+				if (!this.mqttClient.isConnected()) {
```

Comment:
Right. I believe the same `this.mqttClient = getClientManager().getClient()` would apply in this passive component as well.
So, if you don't provide auto-reconnect to the `ClientManager` options, then client would never be connected. 
And that is not so bad to let this passive component to reconnect on message produced.

Reply:
Yeah, I will add this event.
Initially my thoughts were: the user will still receive this 'failed' event in case of not successful delivery / subscription etc. But why wait for error in the logic when we can 'warn' the user earlier.